In [4]:
import pandas as pd
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
from sklearn.svm import SVC

In [5]:
names = ['sepal length', 'sepal width', 'petal lengt', 'petal width', 'class']
data = pd.read_csv("iris.data", names=names)

In [14]:
data_without_class=data.drop(columns = ['class'],axis = 1)
array = data.values

In [17]:
X = array[:,0:4]
Y = array[:,4]

# **Logistic Regression**

In [23]:
kfold = KFold(n_splits=10, random_state=7, shuffle=True)
model = LogisticRegression(solver='liblinear')
results = cross_val_score(model, X, Y, cv=kfold)
print(f"{round(results.mean()*100, 2)} %")

95.33 %


Значение точности вычислений составляет 95%, модель логистической регрессии хорошо подходит для классификации.

# **Support Vector Machines**

In [21]:
kfold = KFold(n_splits=10, random_state=7, shuffle=True)
model = SVC()
results = cross_val_score(model, X, Y, cv=kfold)
print(f"{round(results.mean()*100, 2)} %")

96.0 %


Линейный метод опорных векторов показал точность классификации 96%, лучше логистической регрессии, но хуже линейного дискриминантного анализа

# **Linear Discriminant Analysis**

In [22]:
kfold = KFold(n_splits=10, random_state=7, shuffle=True)
model = LinearDiscriminantAnalysis()
results = cross_val_score(model, X, Y, cv=kfold)
print(f"{round(results.mean()*100, 2)} %")

98.0 %


Линейный дискриминантный анализ показал точность классификации 98%, показатель лучше, чем логистическая регрессия

Таким образом, лучше всего подходит модель линейного дискриминантного анализа